### PR TITLE
feat: support setting an HTTP user agent

### PR DESCRIPTION
**Pull request by velocitysystems**
Refs #61.

No user agent was set anywhere, so each transport sent its own: nothing on desktop, `okhttp/4.12.0` on Android, `<app>/<version> CFNetwork/…` on iOS. A plugin builder now sets one for all three:

```rust
tauri_plugin_download::Builder::new()
   .user_agent("my-app/1.0")
   .build()
```

`init()` stays as shorthand for `Builder::new().build()`. Left unset, each platform keeps what it sends today.

### Getting the value to the native plugins

Tauri builds the config it forwards to a native `load()` from `tauri.conf.json` (`tauri-2.9.5/src/plugin.rs:799`), so a value set on the Rust builder cannot travel that channel. It is pushed with a `configure` command from plugin `setup()` instead — after registration, and before the event loop exists, so before the webview can invoke anything. `run_mobile_plugin` bypasses the ACL, so `configure` needs no `COMMANDS` entry or permission and is unreachable from JS.

Raised upstream as tauri-apps/tauri#15925, proposing that `PluginApi` gain a way to pass builder-supplied config to the native side. If that lands, the `configure` command can be dropped from both Kotlin and Swift and the value read through the existing `getConfig` / `parseConfig` instead.

Desktop configures its one `reqwest` client. iOS sets the header per request, because the background session is built in `init()` before any value can arrive. Android captures it into the WorkManager input data at enqueue time — WorkManager can re-run a stranded worker in a process where the plugin never loads, so an in-memory value would be lost.

### Validation is printable ASCII, not the full header spec

`HeaderValue` accepts the UTF-8 bytes of non-ASCII characters; OkHttp's `checkValue` does not. So `.user_agent("Café/1.0")` would pass startup, work on desktop and iOS, and throw in `DownloadWorker` on every Android download — into the permanent-failure branch, since `IllegalArgumentException` is not an `IOException`. Confirmed against the project's own OkHttp: `Unexpected char 0xe9 at 3`. Empty is accepted, being legal HTTP.

### Notes for review

* A changed user agent reaches in-flight downloads differently per platform: fixed at enqueue time on Android and at start on iOS, while desktop's one-client-per-process means a resumed download picks up the new value. All three are in the README.
* The Kotlin bridge is compile-checked nowhere — it needs `android/.tauri/tauri-api`, which only exists during a real app build, and CI runs `:lib:test`. Inspection-verified only.
* Not run on device. The case worth exercising is force-stopping mid-download on Android and letting WorkManager re-run the worker.
* `await userAgentHolder.value` widens a pre-existing iOS window between the `.inProgress` commit and task creation, where a concurrent `cancel()` orphans the download. Not new — `await store.update(record)` was already a suspension point there. Closing it means serialising start/cancel per path, which is not this change's job.

### CODE REVIEW

File: android/lib/src/main/java/org/silvermine/downloadmanager/DownloadWorker.kt [hunk 1]
```diff
@@ -62,6 +62,12 @@ internal class DownloadWorker(
 
          // Build request with Range header for resuming.
          val requestBuilder = Request.Builder().url(url)
+         // Read from the input data rather than the manager: a worker re-run after
+         // process death has no loaded plugin to have set it. Absent leaves OkHttp's
+         // own default in place.
+         inputData.getString(KEY_USER_AGENT)?.let {
```
Comment (jjhafer) [inline android/lib/src/main/java/org/silvermine/downloadmanager/DownloadWorker.kt]: The only Android code that sets the User-Agent header has no test coverage — deleting it, or wiring it to the wrong input key, still leaves CI green.

<details>
<summary>Details</summary>

Deleting the whole `?.let` block below and running `./gradlew :lib:test` still exits 0. Two further mutations also survive: hardcoding a wrong constant instead of reading input data, and reading the wrong input-data key (`KEY_URL` instead of `KEY_USER_AGENT`). So the Android half of this feature can be entirely absent, or silently sending the download URL as the user agent, with all three CI checks green.

```kotlin
inputData.getString(KEY_USER_AGENT)?.let {
   requestBuilder.header("User-Agent", it)
}
```

`DownloadWorkerTest.kt` only covers attempt counting — its own comment explains why: a `CoroutineWorker` cannot be built without WorkManager's test artifact. Compare the desktop side, which is genuinely well covered because the four new wiremock tests assert on what the server actually received.

Secondary, in the sibling test file: the test named `no user agent leaves the key absent` (`DownloadManagerTest.kt`) is wrong about the mechanism. Measured against work-runtime 2.9.1, `workDataOf(KEY_USER_AGENT to null)` stores the key WITH a null value — `size()==3`, `getKeyValueMap().containsKey(...)==true` — and it round-trips through `toByteArrayInternal` intact. The `assertNull` assertion itself is correct; only the name and comment mislead.

What do you think about hoisting the request construction into an internal companion function next to `inputDataFor`/`constraintsFor` — e.g. `internal fun requestFor(url: String, userAgent: String?, downloadedSize: Long): Request` — and asserting on it directly (header set, header absent, and ideally that Range and User-Agent coexist, mirroring the existing Rust `test_user_agent_and_range_header_are_both_sent_on_resume`)? OkHttp is already on `:lib`'s unit-test classpath, so this needs no new dependency. Also worth renaming the sibling test to something like `no user agent stores a null value`.

</details>

*Comment added by AI model Claude Sonnet*

File: src/lib.rs [hunk 2]
```diff
@@ -56,56 +56,108 @@ impl<R: Runtime, T: Manager<R>> crate::DownloadExt<R> for T {
...
+            commands::cancel,
+            commands::pause,
+            commands::resume,
+            commands::is_native,
+         ])
+         .setup(move |app, _api| {
+            // Validated before either platform branch so an invalid value fails the
+            // same way everywhere, rather than only where the transport rejects it.
+            if let Some(ref user_agent) = user_agent {
+               download_manager::validate_user_agent(user_agent)?;
```
Comment (jjhafer) [inline src/lib.rs]: The plugin's validation call has no test pinning it, so the README's central promise about it is currently unverified — worth adding a couple of small tests.

<details>
<summary>Details</summary>

Deleting this validation call leaves the whole Rust suite green (125 tests pass). `src/models.rs` is the only file under `src/` with a `#[cfg(test)]` block, so the entire plugin `Builder` is unpinned — which means `Builder::user_agent` could also discard its argument entirely (`let _ = user_agent;`) with every test still passing, on all three platforms.

```rust
download_manager::validate_user_agent(user_agent)?;
```

The two claims currently resting on untested code are README.md:152-153 ("An invalid value fails plugin initialization rather than surfacing later as a failed download on one platform") and the same sentence in the `Builder::user_agent` doc comment.

Two cheap tests would close this gap. For the setter, a `#[cfg(test)] mod tests` in `src/lib.rs` asserting `Builder::new().user_agent("my-app/1.0").user_agent == Some("my-app/1.0".to_string())` is about two lines. For the validation promise, tauri ships `tauri::test::mock_builder` behind its `test` feature; adding it as a dev-dependency would let you assert that `Builder::new().user_agent("Café/1.0").build()` fails app construction with a `PluginInitialization` error.

</details>

*Comment added by AI model Claude Sonnet*

File: src/mobile.rs [hunk 3]
```diff
@@ -13,18 +13,55 @@ tauri::ios_plugin_binding!(init_plugin_download);
...
 ) -> crate::Result<Download<R>> {
    #[cfg(target_os = "android")]
    let handle = _api.register_android_plugin(PLUGIN_IDENTIFIER, "DownloadPlugin")?;
    #[cfg(target_os = "ios")]
    let handle = _api.register_ios_plugin(init_plugin_download)?;
-   Ok(Download(handle))
+
+   let download = Download(handle);
+
+   // Pushed rather than carried by the plugin config: Tauri fills the config it
```
Comment (jjhafer) [inline src/mobile.rs]: The comment gets the mechanism backwards — omission from `generate_handler!` is what *routes* `configure` to the native plugin, not what blocks it, so the real safety net (the ACL) deserves the comment.

<details>
<summary>Details</summary>

The conclusion here (and in the PR description) is correct — `configure` really is unreachable from JS — but the stated mechanism is inverted, and the inversion hides the invariant that actually protects it.

```rust
// Pushed rather than carried by the plugin config: Tauri fills the config it
// forwards to `load()` from `tauri.conf.json`, so a value set on the Rust
// builder has no way into it.
```

On mobile, `tauri/src/webview/mod.rs:1842-1846` has a `#[cfg(mobile)]` fallback: after `let mut handled = manager.extend_api(plugin, invoke)`, any `plugin:download|<cmd>` that NO Rust command claims is forwarded straight to the native plugin via `plugin::mobile::run_command`, lower-camel-cased. So absence from `tauri::generate_handler!` is precisely what ROUTES a JS call to the Kotlin/Swift handler, rather than what blocks it. This repo already depends on that fallback: `registerListener` is in `build.rs` COMMANDS and `permissions/default.toml` but has no implementation anywhere in this repository — it works only because the fallback delivers it to Tauri's native `Plugin` base class.

The thing that actually stops a webview reaching `configure` is the ACL gate at `webview/mod.rs:1801-1829` plus `RuntimeAuthority::resolve_access` (`ipc/authority.rs:439-470`), which is an exact map lookup with no wildcards — `configure` has no entry, so `invoke.acl` is `None` and the call is rejected. Consequence: the load-bearing invariant is "never add `configure` to `build.rs` COMMANDS", not "never add it to `generate_handler!`". Adding it to COMMANDS auto-generates `allow-configure`, and if that ever lands in the default permission set, the webview could call the native `configure` directly with an arbitrary `userAgent` — a handler that validates nothing, since `validate_user_agent` lives only on the Rust route. A future reviewer asking "shouldn't `configure` have a permission entry for consistency with the other commands?" would be asking for exactly the wrong thing.

What do you think about rewording the comment to name the ACL as the gate rather than `run_mobile_plugin`'s bypass, and adding a short note above COMMANDS in `build.rs` saying `configure` must never be listed there and why? Two smaller precision fixes in the same comment: "block until the native plugin is loaded" should be "registered" — with no webview in existence at setup time, `register_ios_plugin` takes its else branch and `load(webview:)` has not run — and the guarantee is stronger than "the webview cannot invoke a command until setup returns", since `initialize_plugins` runs in `Builder::build` (`app.rs:2291`) while webviews are created later in `App::run` (`app.rs:1300`), so no webview object exists at all yet.

</details>

*Comment added by AI model Claude Sonnet*

File: src/mobile.rs [hunk 4]
```diff
@@ -13,18 +13,55 @@ tauri::ios_plugin_binding!(init_plugin_download);
...
+   // forwards to `load()` from `tauri.conf.json`, so a value set on the Rust
+   // builder has no way into it. Both registrations above block until the native
+   // plugin is loaded, and the webview cannot invoke a command until setup returns,
+   // so this lands before any download can be created.
+   if let Some(user_agent) = user_agent {
+      download.configure(&user_agent)?;
```
Comment (jjhafer) [inline src/mobile.rs]: Propagating this error with `?` makes any native transport failure fatal to app launch — is that the intended tradeoff, given the only thing being protected is the download user-agent header?

<details>
<summary>Details</summary>

Validation has already run at `src/lib.rs:113`, so every remaining way this line can fail is transport: a Jackson or R8 field-name mismatch, an exception in the Kotlin handler, a JNI error, or using the wrong `invoke.resolve()` overload. Each becomes `tauri::Error::PluginInitialization`, so `tauri::Builder::run` returns `Err` and the consumer's `.expect(...)` panics — the app does not start.

```rust
download.configure(&user_agent)?;
```

The consequence being avoided is "downloads send the platform's default user agent". It's also slightly worse than a clean error: `run_mobile_plugin` does `rx.recv().unwrap()` (`tauri/src/plugin/mobile.rs:337`), so a dropped sender panics inside plugin setup rather than returning `Err`. This matters more than usual because no CI job compiles either native bridge (both need the absent `.tauri/tauri-api`), so a typo in the Kotlin/Swift command name or in the `userAgent` key would first surface as "the app does not launch on Android/iOS", and only for consumers who configured a user agent.

Worth noting if you consider a log-and-continue approach: both this crate and download-manager enable tracing's `release_max_level_off`, so a `warn!` here compiles to nothing in a release build — the honest options are abort-launch or silently ignore, not "warn and continue".

If launch-abort is deliberate, a sentence saying so would help, since fail-fast on validation (deterministic, caught on a developer's machine) and fail-fast on transport (nondeterministic, field-only) are different calls. If not, handling the `Err` here and continuing — accepting that the diagnostic is release-silent, or emitting it some other way — would keep the validation benefit without tying app launch to an HTTP header.

</details>

*Comment added by AI model Claude Sonnet*

File: crates/download-manager/src/validate.rs [hunk 5]
```diff
@@ -59,6 +59,33 @@ pub fn url(url: &str) -> crate::Result<()> {
...
+/// transports accept.
+///
+/// Narrower than `HeaderValue`, which reqwest would apply on its own: that admits any
+/// byte from `0x20` up except `0x7F`, so the UTF-8 bytes of a non-ASCII character pass
+/// it. OkHttp rejects them, failing every Android download when the request is built.
+///
+/// Unlike [`path`] and [`url`], there is no emptiness check: an empty user agent is
+/// legal HTTP, and rejecting it would be this crate's policy rather than the
+/// specification's.
+pub fn user_agent(user_agent: &str) -> crate::Result<()> {
```
Comment (jjhafer) [inline crates/download-manager/src/validate.rs]: The user-agent validator has two completeness gaps worth closing: the horizontal-tab allowance is untested (and matches OkHttp intentionally), and there's no length bound, which lets a WorkManager 10 KB limit surprise Android alone.

<details>
<summary>Details</summary>

(1) The horizontal-tab allowance is untested and is a live mutation survivor: deleting `c != '\t' &&` leaves all 13 validate tests passing, while deleting `c != ' ' &&` is correctly caught. The tab behaviour is also the one place the docs are stricter than the code — README.md:151 and the `Builder::user_agent` doc comment both say "must be printable ASCII", but tab is not printable ASCII and is currently accepted. This function's own doc comment ("Accepts horizontal tab and printable ASCII") is the accurate one, and it is right to accept tab: OkHttp 4.12.0's `Headers.checkValue` accepts a char iff `c == 0x09 || (0x20 <= c < 0x7F)`, so the accepted set here matches OkHttp exactly.

```rust
pub fn user_agent(user_agent: &str) -> crate::Result<()> {
```

(2) There is no length bound, which gives Android an upper limit the rest of this design exists to eliminate. Measured against work-runtime 2.9.1: `Data.MAX_DATA_BYTES` is 10240 and `Data.Builder.build()` throws `IllegalStateException` ("Data cannot occupy more than 10240 bytes when serialized"); with a representative url and path, a user agent of about 10,110 characters is enough. The throw happens inside `inputDataFor`, i.e. inside `enqueueDownload`, so every `start()`/`resume()` fails on Android with an opaque WorkManager message while desktop and iOS accept the same value — exactly the one-platform divergence README.md:152-153 says the design prevents. The Data budget is shared with `url` and `path`, so the real limit is lower still.

What do you think about adding one assertion for tab in `test_valid_user_agents`, e.g. `assert!(user_agent("my-app/1.0 (build\t1)").is_ok());`, and rewording README.md:151 and the `Builder::user_agent` doc to match this function's own wording? For the length, a cap here (1 KB is generous — servers commonly cap a header line at 8 KB) would make an over-long value fail at plugin init on all three platforms instead of at enqueue on one.

</details>

*Comment added by AI model Claude Sonnet*

File: crates/download-manager/src/manager.rs [hunk 6]
```diff
@@ -61,7 +72,25 @@ impl DownloadManager {
...
-      let http_client = ClientBuilder::new(reqwest::Client::new())
+
+      let mut client_builder = reqwest::Client::builder();
+      if let Some(ref user_agent) = config.user_agent {
+         // Checked again here rather than trusted: this is a public constructor on a
+         // Tauri-agnostic crate, so it cannot assume the caller validated first, and
+         // an unusable value would otherwise panic the build below.
+         match validate::user_agent(user_agent) {
+            Ok(()) => client_builder = client_builder.user_agent(user_agent.clone()),
+            Err(e) => warn!("Ignoring invalid user agent: {}", e),
```
Comment (jjhafer) [inline crates/download-manager/src/manager.rs]: The comment says an unusable user agent is dropped "with a warning", but this crate compiles out `warn!` in release builds, so that diagnostic never actually fires.

<details>
<summary>Details</summary>

`crates/download-manager/Cargo.toml:17` enables tracing's `release_max_level_off`, and `tracing-0.1.44/src/level_filters.rs:68-73` sets `STATIC_MAX_LEVEL = LevelFilter::OFF` under `not(debug_assertions)`, so this `warn!` is compile-time elided in release builds.

```rust
Err(e) => warn!("Ignoring invalid user agent: {}", e),
```

The rest of this block is accurate — I checked reqwest 0.13.1, where `ClientBuilder::user_agent` stores the conversion error into `config.error` for `build()` to surface, and `Client::new()` really is `ClientBuilder::new().build().expect("Client::new()")`, so both the "would otherwise panic the build below" and the "mirrors `Client::new()`" comments are exactly right. The only mismatch is that in the shipping configuration, a direct download-manager consumer who passes an unusable value gets no user agent and no signal, while the same value hard-fails through the plugin. The test name `test_an_unusable_user_agent_is_dropped_rather_than_panicking` is accurate; the comment's "with a warning" is not.

What do you think about either noting in the comment that the diagnostic is elided in release given this crate's tracing configuration, or making `DownloadManager::new` fallible and returning the error — it has one in-repo caller, and the setup closure it is called from already returns `Result`?

</details>

*Comment added by AI model Claude Sonnet*

File: README.md [hunk 7]
```diff
@@ -115,6 +115,55 @@ fn main() {
...
+
+A download that outlives the process picks up a changed user agent differently on each
+platform:
+
+   * **Desktop** builds one HTTP client per process, so a download resumed after a
+     relaunch uses the new run's value, not the one it started with.
+   * **Android** captures the value in the work request when a download is enqueued,
+     because WorkManager can re-run a stranded worker in a process where the plugin
+     never loads. A change applies only to downloads enqueued after it.
+   * **iOS** resumes from resume data, which carries the original request, so a resumed
```
Comment (jjhafer) [inline README.md]: The iOS resume bullet is stated unconditionally, but the code only keeps the original user agent when resume data actually exists — worth qualifying, along with a related overstatement one line above about all three platforms sending "the same value".

<details>
<summary>Details</summary>

(1) `DownloadManager.swift:206-226`: the else branch — taken when resume data is absent or empty — calls `Self.request(for: record, userAgent: await userAgentHolder.value)` at :224, i.e. the CURRENT value, and resets received bytes to zero. That branch isn't exotic; the code `os_log`s it at :214-216 and the comment at :198-202 explains that a server without byte-range support leaves a paused record with no resume data. So iOS keeps the value it started with only when resume data exists.

```markdown
   * **iOS** resumes from resume data, which carries the original request, so a resumed
     download keeps the value it started with.
```

(2) README.md:151 "Setting it makes all three send the same value" isn't true for whitespace-padded values. Measured on OkHttp 4.12.0, header values are trimmed: `" my-app/1.0 "` arrives as `"my-app/1.0"` and `"   "` arrives as `""`. Desktop (http 1.4.0's `HeaderValue`) and Foundation both send the value verbatim, so a padded user agent differs on Android. Since `validate::user_agent` deliberately accepts space, this is reachable.

(3) Optional, for the Android bullet at :161-163: it's literally true, but positioned against the desktop bullet it reads as the opposite of the actual behaviour. `resume()` calls `enqueueDownload` again (`DownloadManager.kt:171`), which reads the current volatile field, so an explicit resume after a relaunch uses the NEW value just like desktop. Only WorkManager auto-re-running a stranded worker keeps the captured one.

What do you think about adding one clause to the iOS bullet ("...when resume data exists; without it the download restarts from zero and picks up the current value"), and either trimming leading/trailing whitespace in `validate::user_agent` or noting the Android exception next to the "same value" sentence? For the Android bullet, distinguishing an explicit resume from a WorkManager auto-retry would make the three-way contrast match the code.

</details>

*Comment added by AI model Claude Sonnet*

File: ios/DownloadManagerKit/Sources/DownloadManagerKit/DownloadManager.swift [hunk 8]
```diff
@@ -127,6 +129,19 @@ public final class DownloadManager: NSObject {
...
+   /**
+    Sets the user agent sent with every download request.
+
+    Applied per request rather than on the session configuration: the background
+    session is built in `init()`, before any value can arrive, and is not something
+    to recreate at runtime. Same reason the network policy is per request.
+
+    - Parameter userAgent: The user agent, or `nil` to leave URLSession's default.
+    */
+   public func setUserAgent(_ userAgent: String?) async {
```
Comment (jjhafer) [inline ios/DownloadManagerKit/Sources/DownloadManagerKit/DownloadManager.swift]: Nothing tests `setUserAgent`/`UserAgentHolder`, so this method could be a silent no-op — a small actor test would catch that, and there's a related ordering hazard in the plugin worth a guarding comment.

<details>
<summary>Details</summary>

Nothing anywhere references `setUserAgent`, `UserAgentHolder`, or `userAgentHolder` from a test, so this method can be a complete no-op — and `UserAgentHolder.set` can store `nil`, and both request-building call sites can pass `nil` — with all 67 Swift tests passing. `request(for:userAgent:)` itself is well covered by the two new tests; it's the holder and the wiring into it that are unpinned.

```swift
public func setUserAgent(_ userAgent: String?) async {
```

That matters more on iOS than elsewhere because `ios/Sources/DownloadPlugin.swift` is compiled by no CI job at all (`ios/Package.swift` needs the absent `../.tauri/tauri-api`), and because the failure is silent: downloads still succeed, just with the wrong user agent.

Related fragility in the same feature, worth a comment even if you don't add a test: in `ios/Sources/DownloadPlugin.swift` the `invoke.resolve()` sits INSIDE the `Task`, after the `await`. That placement is load-bearing — moving it outside would let Rust's blocking `run_mobile_plugin` return before the actor write lands, silently voiding the before-any-download-exists ordering the whole design rests on — and no compiler or test in this repo would notice.

One unrelated small thing in the new file: `UserAgentHolder.swift:10` uses `[DownloadManager]`, which is KDoc/Markdown link syntax and renders literally in Swift Quick Help; it's the only bare `[Symbol]` in the Swift tree, which otherwise uses backticks — including this method's own doc comment two lines above.

What do you think about a three-line test on the actor (set a value, read it back) to kill two of the three surviving mutations? The third — that `start()` and `resume()` actually pass the holder's value — needs test infrastructure this package doesn't have, so naming it as a known gap seems reasonable. Also, adding a short comment in `DownloadPlugin.swift`'s `configure` saying `invoke.resolve()` must stay inside the `Task`, and changing `[DownloadManager]` to backticks, would be quick wins.

</details>

*Comment added by AI model Claude Sonnet*

File: android/lib/src/main/java/org/silvermine/downloadmanager/DownloadWorker.kt (hunk 1 above)
Comment (velocitysystems) [reply android/lib/src/main/java/org/silvermine/downloadmanager/DownloadWorker.kt]: ✅ **Applied** — **Extracted `requestFor` into the worker's companion, with four tests**

Put it in `DownloadWorker.kt`'s companion rather than beside `inputDataFor`, since `DownloadManager.kt` has no OkHttp coupling. Tests cover the header set, absent, coexisting with `Range`, and `Range` absent at zero bytes. Dropping the header now fails two of them; widening `> 0` to `>= 0` fails another. The one-line `KEY_USER_AGENT` read in `doWork` stays a bare delegation and uncovered.

Renamed the sibling test too — I measured it: `workDataOf(k to null)` gives `size=3`, `containsKey=true`. You were right that the name described the wrong mechanism.

*Comment added by AI model Claude*

File: src/lib.rs (hunk 2 above)
Comment (velocitysystems) [reply src/lib.rs]: ✅ **Applied** — **Added `Builder` tests, including the init-failure one via `mock_builder`**

Confirmed your mutation first: replacing the call with `let _ = user_agent;` left everything green. `tauri`'s `test` feature is empty, so the dev-dependency pulls in nothing new and `Cargo.lock` is unchanged.

Kept the setter test and both `mock_builder` tests, and dropped a third asserting `new()` leaves it `None` — it killed no plausible mutation. `test_an_invalid_user_agent_fails_plugin_initialization` fails with the validation removed and passes with it restored, so the README's promise is now pinned.

*Comment added by AI model Claude*

File: src/mobile.rs (hunk 3 above)
Comment (velocitysystems) [reply src/mobile.rs]: ✅ **Applied** — **Comment now names the ACL as the gate**

You're right, and I verified it against 2.9.5: the ACL check at `webview/mod.rs:1802` rejects and returns before the `#[cfg(mobile)]` fallback at `:1844` forwards unclaimed commands to native. So absence from `generate_handler!` routes rather than blocks — `registerListener` is exactly that, in COMMANDS with no Rust implementation in this repo.

Also corrected "loaded" → "registered" and the ordering claim. The comment now records the invariant that `configure` must never enter COMMANDS; I left `build.rs` itself untouched.

*Comment added by AI model Claude*

File: src/mobile.rs (hunk 4 above)
Comment (velocitysystems) [reply src/mobile.rs]: ✅ **Applied (modified)** — **Documented the launch-abort as deliberate, scoped to what actually reaches it**

Behaviour unchanged. Your `release_max_level_off` point is what settles it: `warn!` is elided in release, so "log and continue" means shipping the wrong user agent silently — worse than failing loudly.

I scoped the comment to that rather than to the failure list, since Jackson/R8 mismatches and exceptions in the Kotlin handler are caught by `PluginManager.dispatchPluginMessage` and arrive as a clean `Err`. The `rx.recv().unwrap()` tail is real but shared by all seven `run_mobile_plugin` calls in the file.

*Comment added by AI model Claude*

File: crates/download-manager/src/validate.rs (hunk 5 above)
Comment (velocitysystems) [reply crates/download-manager/src/validate.rs]: ✅ **Partially applied** — **Tab test and doc fix applied; length cap declined**

Tab: applied. It was a live survivor, and the README and the `Builder::user_agent` doc did contradict the function's own comment. Both now say printable ASCII or horizontal tab.

Length: declined. A user-agent-only cap needs a hand-picked constant and doesn't close the divergence you describe — `url` and `path` share the same 10 KB `Data` budget and are equally unbounded, so an oversized URL still fails on Android alone. The fix that works is a size check at the enqueue site against `Data.MAX_DATA_BYTES`, covering all three. Worth its own issue.

*Comment added by AI model Claude*

File: crates/download-manager/src/manager.rs (hunk 6 above)
Comment (velocitysystems) [reply crates/download-manager/src/manager.rs]: ✅ **Applied (modified)** — **Comment now says the diagnostic is elided in release; signature left alone**

Took (a). Confirmed `STATIC_MAX_LEVEL` resolves to `OFF` under `not(debug_assertions)` with `release_max_level_off`, which both crates set.

Declined (b): making `new` fallible wouldn't restore visibility, because the caller's own `warn!` is elided by the same feature — and it would invert the posture of the `store.load()` failure three lines above, which swallows and warns identically.

*Comment added by AI model Claude*

File: README.md (hunk 7 above)
Comment (velocitysystems) [reply README.md]: ✅ **Partially applied** — **(1) and (3) applied; (2) declined after measuring it**

(1) and (3) as you described: iOS keeps the original value only when resume data exists, and Android reads the current value on an explicit `resume` — only a WorkManager auto-rerun uses the captured one. Both are in the README now.

(2) I first made `validate::user_agent` reject padded values, then reverted it. I sent `" my-app/1.0 "` through reqwest to a real server and it arrived as `my-app/1.0`: recipients strip leading and trailing whitespace (RFC 9110 §5.5). Android trims client-side, the other two send padding the server discards — so all three deliver the same bytes and there is nothing to fix.

*Comment added by AI model Claude*

File: ios/DownloadManagerKit/Sources/DownloadManagerKit/DownloadManager.swift (hunk 8 above)
Comment (velocitysystems) [reply ios/DownloadManagerKit/Sources/DownloadManagerKit/DownloadManager.swift]: ✅ **Applied** — **Two actor tests, the `configure` comment, and the autolink form**

The holder test isn't tautological — making `set` a no-op now fails it. Added a second for overwrite/clear, since a holder that latched its first value would pass the first.

For `resolve()`, I scoped the comment to why `configure` is the one handler where hoisting it out would still compile: every other is anchored by a response payload it needs. Changed `[DownloadManager]` to `` [`DownloadManager`] ``, matching `DownloadRecord.swift:11`. The `start`/`resume` wiring is flagged as a known gap on `setUserAgent`.

*Comment added by AI model Claude*

**Review by jjhafer — APPROVED** [review]
